spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/BaseCassandraRDD.scala

```diff
@@ -0,0 +1,245 @@
+  *
+  * By default, reads are performed at ConsistencyLevel.LOCAL_ONE in order to leverage data-locality and minimize network traffic.
+  * This read consistency level is controlled by the following property:
+  *
+  * - spark.cassandra.input.consistency.level: consistency level for RDD reads, string matching the ConsistencyLevel enum name.
+  *
+  * If a Cassandra node fails or gets overloaded during read, queries are retried to a different node.
+  */
+abstract class BaseCassandraRDD[R, P] private[connector](
+                                                          @transient sc: SparkContext,
+                                                          val connector: CassandraConnector,
+                                                          val keyspaceName: String,
+                                                          val tableName: String,
+                                                          val columnNames: ColumnSelector = AllColumns,
+                                                          val where: CqlWhereClause = CqlWhereClause.empty,
+                                                          val readConf: ReadConf = ReadConf(),
+                                                          val dependecyList: Seq[Dependency[_]] = Seq.empty)(
+                                                          implicit
+                                                          ct: ClassTag[R], rddt: ClassTag[P],
+                                                          @transient rtf: RowReaderFactory[R])
+  extends RDD[P](sc, dependecyList) with Logging {
+
+  /* Logging classes inheritance conflict fix. */
+  override def log = super[Logging].log
+
+  override def logName = super[Logging].logName
+
+  override def logInfo(msg: => String) = super[Logging].logInfo(msg)
+
+  override def logDebug(msg: => String) = super[Logging].logDebug(msg)
+
+  override def logTrace(msg: => String) = super[Logging].logTrace(msg)
+
+  override def logWarning(msg: => String) = super[Logging].logWarning(msg)
+
+  override def logError(msg: => String) = super[Logging].logError(msg)
+
+  override def logInfo(msg: => String, throwable: Throwable) = super[Logging].logInfo(msg, throwable)
+
+  override def logDebug(msg: => String, throwable: Throwable) = super[Logging].logDebug(msg, throwable)
+
+  override def logTrace(msg: => String, throwable: Throwable) = super[Logging].logTrace(msg, throwable)
+
+  override def logWarning(msg: => String, throwable: Throwable) = super[Logging].logWarning(msg, throwable)
+
+  override def logError(msg: => String, throwable: Throwable) = super[Logging].logError(msg, throwable)
+
+  override def isTraceEnabled() = super[Logging].isTraceEnabled()
+
+  protected def fetchSize = readConf.fetchSize
+
+  protected def splitSize = readConf.splitSize
+
+  protected def consistencyLevel = readConf.consistencyLevel
+
+  protected def copy(columnNames: ColumnSelector = columnNames,
```

Comment:
I suppose that you may want to try using `this.type` as a return type. Take a look [here](http://scalada.blogspot.com/2008/02/thistype-for-chaining-method-calls.html)